tests/dummy/app/services/dynamic-chart.js

```diff
@@ -0,0 +1,28 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+
+  getRandomInt(min, max) {
+    return parseInt(Math.random() * (max - min) + min, 10);
+  },
+
+  updateSeriesData(chartData, rangeStart, rangeEnd) {
+    console.log('updateSeriesData');
+
+    let numPoints = this.getRandomInt(rangeStart, rangeEnd);
+    let chartDataCopy = Ember.copy(chartData, true);
+
+    chartDataCopy.forEach(series => {
```

Comment:
Good point. Map takes care of the cloning instead of using `Ember.copy`. I updated the code to use map.